public/docs/_examples/structural-directives/dart/lib/heavy_loader_component.dart

```diff
@@ -0,0 +1,37 @@
+// #docregion
+import 'package:angular2/angular2.dart';
+import 'dart:async';
+
+int nextId = 1;
+
+@Component(
+    selector: 'heavy-loader',
+    template: '<span>heavy loader #{{id}} on duty!</span>')
+class HeavyLoaderComponent implements OnInit, OnDestroy {
+  int id = nextId++;
+  @Input() List<String> logs;
+
+  ngOnInit() {
+    // Mock todo: get 10,000 rows of data from the server
+    _log(
+        "heavy-loader ${id} initialized, loading 10,000 rows of data from the server");
+  }
+
+  ngOnDestroy() {
+    // Mock todo: clean-up
+    _log("heavy-loader ${id} destroyed, cleaning up");
+  }
+
+  _log(String msg) {
+    logs.add(msg);
+    _tick();
+  }
+
+  /// Triggers the next round of Angular change detection
+  /// after one turn of the JavaScript cycle
+  /// ensuring display of msg added in onDestroy
+  _tick() {
+    new Future(() {});
```

Comment:
That's just the Dart equivalent of the TS code, which I assume is just to kick the event loop. I have no real knowledge of how it works, and it's possible that without a return value it might be optimized away.

@adaojunior: Does the app behave differently when you comment out this line?